iree/modules/CMakeLists.txt

```diff
@@ -12,5 +12,5 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory(hal)
+add_subdirectory(check)
```

Comment:
Probably want hal as well? Or it doesn't go here because it's at the root? Can it be both? I just expect that if you don't add the subdirectory you might get weird behavior if you try to build/test subsets of the project.

Reply:
Hal is included in the root. This is part of our module ordering issue. Because modules/hal is depended by the VM module, it needs to be listed prior, however the other modules depend on iree/tools for compilation.

If we can correct the CMake ordering issue we should be fine but until that is done we will need to split up the HAL from the rest of our submodules.

Reply:
Yeah, this is weird and will interact poorly with `bazel_to_cmake.py`. I'd prefer to figure out why ordering seemed to matter for Rob and keep the `add_subdirectory` calls simple (each level includes subdirs, no looking two levels deep based on conditions).

Reply:
My best guess is either its related to ALWAYSLINK or that the iree/tools directory is the only location where everything is linked together.

Reply:
Thanks for your patience with the CMake build. Looks like this is pretty clean now.